From 278d08f2942f261ae5db1625bb65be35f78e37d0 Mon Sep 17 00:00:00 2001
From: skandigraun <skandigraun@gmail.com>
Date: Fri, 20 Oct 2023 13:06:21 +0200
Subject: [PATCH] commit

Upstream-Status: Inappropriate
---
 toolkit/moz.configure | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

--- ./toolkit/moz.configure.orig	2024-12-26 20:12:05.536834226 +0100
+++ ./toolkit/moz.configure	2024-12-26 20:18:43.137630913 +0100
@@ -215,7 +215,7 @@
     # DWARF data, but build system issues currently prevent Rust from being
     # used in mozglue.)
     if value and (target.kernel == "Linux" or target.kernel == "FreeBSD"):
-        return True
+        return False
 
 
 set_config("MOZ_GECKO_PROFILER_PARSE_ELF", gecko_profiler_parse_elf)
@@ -718,6 +718,7 @@
 # ==============================================================
 @depends(toolkit)
 def pdf_printing(toolkit):
+    return False
     if toolkit in ("windows", "gtk", "android"):
         return True
 
@@ -1048,7 +1049,7 @@
 project_flag(
     "MOZ_SERVICES_HEALTHREPORT",
     help="Build Firefox Health Reporter Service",
-    set_as_define=True,
+    set_as_define=False,
 )
 
 project_flag(
@@ -1126,7 +1127,7 @@
 
 option(
     env="MOZ_TELEMETRY_REPORTING",
-    default=mozilla_official,
+    default=False,
     help="Enable telemetry reporting",
 )
 
@@ -1135,7 +1136,7 @@
 
 @depends("MOZ_TELEMETRY_REPORTING", milestone.is_nightly)
 def telemetry_on_by_default(reporting, is_nightly):
-    return reporting and is_nightly
+    return False
 
 
 set_define("MOZ_TELEMETRY_ON_BY_DEFAULT", True, when=telemetry_on_by_default)
