--- ./src/egl/main/eglapi.c.orig	2024-05-26 14:01:08.294772891 +0200
+++ ./src/egl/main/eglapi.c	2024-05-26 14:01:25.846744330 +0200
@@ -645,7 +645,6 @@
        disp->Extensions.KHR_gl_texture_cubemap_image &&
        disp->Extensions.KHR_gl_renderbuffer_image &&
        disp->Extensions.KHR_create_context &&
-       disp->Extensions.EXT_create_context_robustness &&
        disp->Extensions.KHR_get_all_proc_addresses &&
        disp->Extensions.KHR_gl_colorspace &&
        disp->Extensions.KHR_surfaceless_context)
