Upstream-Status: Pending
--- ./src/egl/drivers/dri2/egl_dri2.c.orig	2024-05-30 20:42:57.115345919 +0200
+++ ./src/egl/drivers/dri2/egl_dri2.c	2024-05-30 20:44:23.408008915 +0200
@@ -1388,8 +1388,15 @@
        *     the requested profile, then an EGL_BAD_MATCH error is generated.
        */
    case __DRI_CTX_ERROR_BAD_API:
+      _eglLog(_EGL_WARNING, "XXX: BAD_API");
+      egl_error = EGL_BAD_MATCH;
+      break;
    case __DRI_CTX_ERROR_BAD_VERSION:
+      _eglLog(_EGL_WARNING, "XXX: BAD_VERSION");
+      egl_error = EGL_BAD_MATCH;
+      break;
    case __DRI_CTX_ERROR_BAD_FLAG:
+      _eglLog(_EGL_WARNING, "XXX: BAD_FLAG");
       egl_error = EGL_BAD_MATCH;
       break;
 
--- ./src/gallium/frontends/dri/dri_context.c.orig	2024-05-30 21:00:53.639341999 +0200
+++ ./src/gallium/frontends/dri/dri_context.c	2024-05-30 21:02:54.206874761 +0200
@@ -186,6 +186,7 @@
 	 *error = __DRI_CTX_ERROR_NO_MEMORY;
 	 break;
       case ST_CONTEXT_ERROR_BAD_VERSION:
+         fprintf(stderr, "XXX dri_context.c: bad version\n");
 	 *error = __DRI_CTX_ERROR_BAD_VERSION;
 	 break;
       }
--- ./src/gallium/frontends/dri/dri_util.c.orig	2024-05-30 21:03:19.426777029 +0200
+++ ./src/gallium/frontends/dri/dri_util.c	2024-05-30 21:05:21.384319393 +0200
@@ -435,11 +435,15 @@
       return __DRI_CTX_ERROR_BAD_API;
    }
 
-   if (max_version == 0)
+   if (max_version == 0){
+      fprintf(stderr, "XXX dri_util.c: max_version is 0! BAD_VERSION\n");
       return __DRI_CTX_ERROR_BAD_VERSION;
+   }
 
-   if (req_version > max_version)
+   if (req_version > max_version) {
+      fprintf(stderr, "XXX dri_util.c: req_version: %d, max_version: %d, req > max, BAD_VERSION\n", req_version, max_version);
       return __DRI_CTX_ERROR_BAD_VERSION;
+   }
 
    return __DRI_CTX_ERROR_SUCCESS;
 }
--- ./src/gallium/frontends/dri/dri_util.c.orig	2024-05-30 21:14:02.789400226 +0200
+++ ./src/gallium/frontends/dri/dri_util.c	2024-05-30 21:15:08.306158131 +0200
@@ -442,7 +442,8 @@
 
    if (req_version > max_version) {
       fprintf(stderr, "XXX dri_util.c: req_version: %d, max_version: %d, req > max, BAD_VERSION\n", req_version, max_version);
-      return __DRI_CTX_ERROR_BAD_VERSION;
+      fprintf(stderr, "XXX dri_util.c, trying to cheat death...\n");
+      return __DRI_CTX_ERROR_BAD_VERSION;
    }
 
    return __DRI_CTX_ERROR_SUCCESS;
--- ./src/gallium/frontends/dri/dri_util.c.orig	2024-05-31 16:15:11.804441430 +0200
+++ ./src/gallium/frontends/dri/dri_util.c	2024-05-31 16:17:07.088119119 +0200
@@ -557,6 +557,11 @@
         }
     }
 
+    if (ctx_config.major_version == 3 && ctx_config.minor_version == 2) {
+        fprintf(stderr, "Let's see if I break the world...\n");
+        ctx_config.minor_version = 1;
+    }
+
     /* The specific Mesa driver may not support the GL_ARB_compatibilty
      * extension or the compatibility profile.  In that case, we treat an
      * API_OPENGL_COMPAT 3.1 as API_OPENGL_CORE. We reject API_OPENGL_COMPAT
