diff --git a/src/client/qwaylanddisplay.cpp b/src/client/qwaylanddisplay.cpp
index eb43313f..e2f7c019 100644
--- a/src/client/qwaylanddisplay.cpp
+++ b/src/client/qwaylanddisplay.cpp
@@ -635,6 +635,10 @@ void QWaylandDisplay::registry_global(uint32_t id, const QString &interface, uin
         return;
     }
 
+    if (interface == "zwp_text_input_manager_v3"){
+        fprintf(stderr, "XXXXX We got one zwp_text_input_manager_v3!\n");
+    }
+
     if (interface == QLatin1String(QtWayland::wl_output::interface()->name)) {
         mWaitingScreens << mWaylandIntegration->createPlatformScreen(this, version, id);
     } else if (interface == QLatin1String(QtWayland::wl_compositor::interface()->name)) {
@@ -764,6 +768,8 @@ void QWaylandDisplay::registry_global(uint32_t id, const QString &interface, uin
     } else if (
             interface == QLatin1String(QtWayland::qt_toplevel_drag_manager_v1::interface()->name)) {
         mXdgToplevelDragManager.reset(new QtWayland::qt_toplevel_drag_manager_v1(registry, id, 1));
+    } else if (interface == "zwp_text_input_manager_v3"){
+        fprintf(stderr, "XXXXX And.... we dropped it.\n");
     }
 
     mGlobals.append(RegistryGlobal(id, interface, version, registry));
